notus/scanner/loader/gpg_sha_verifier.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+import os
+from pathlib import Path
+from typing import Callable, Dict, Optional
+
+from gnupg import GPG
+
+
+class GPGException(Exception):
+    """Class for exceptions raised in gpg_sha256sums"""
+
+
+def __default_gpg_home() -> GPG:
+    """
+    __defaultGpgHome tries to load the variable 'GPG_HOME' or to guess it
+    """
+    manual = os.getenv("GPG_HOME")
+
+    home = Path(manual) if manual else Path.home() / ".gnupg"
+    return GPG(gnupghome=f"{home.absolute()}")
+
+
+def gpg_sha256sums(
+    hash_file: Path, gpg: Optional[GPG] = None
+) -> Dict[str, str]:
+    """
+    gpg_sha256sums verifies given hash_file with a asc file
+
+    This functions assumes that the asc file is in the same directory as the
+    hashfile and has the same name but with the suffix '.asc'
+    """
+
+    # when doing that via paramater list it is loading eagerly on import
+    # which may fail on some systems
+    if not gpg:
+        gpg = __default_gpg_home()
+    if not hash_file.is_file():
+        raise GPGException(f"{hash_file.absolute()} is not a file")
+    asc_path = hash_file.parent / f"{hash_file.name}.asc"
```

Comment:
I reverted that because asc_path needs to be a path not a string to allow open and Path(f"hash_file.absolute()") is not mockable therefore I prefer parent / "newname"

Reply:
Alternative is to not verify that eagerly and let the actual gnupg module throw an exception; the reasoning for checking eagerly is more optimization based since we use that path to fulfill other requirements but would fail in any case; so that check is a rather weak reason.